massa-execution-worker/src/speculative_ledger.rs

```diff
@@ -535,6 +540,17 @@ impl SpeculativeLedger {
             )));
         }
 
+        // check if caller doesn't have already too many data entries
+        if let Some(keys) = self.get_keys(addr, &[]) {
```

Comment:
in certain cases this might be very heavy but OK for now

Reply:
Added a comment